### PR TITLE
Add the Read/List service functions

### PR DESCRIPTION
**Pull request by talanknight**
Additionally I have set up the handler registration for the worker service so requests can be sent the the `/v1/workers` path.

### CODE REVIEW

File: internal/proto/controller/api/resources/workers/v1/worker.proto [hunk 1]
```diff
@@ -18,9 +18,6 @@ message WorkerConfig {
   // Output only.  The name the worker daemon reports itself having.
   string name = 20;
 
-  // Output only.  The boundary version the worker has reported itself as running.
-  string version = 30;
```
Comment (irenarindos) [inline internal/proto/controller/api/resources/workers/v1/worker.proto:22]: Don't we still need version?

**Review by irenarindos — CHANGES_REQUESTED** [review]

File: internal/servers/testing.go [hunk 2]
```diff
@@ -81,22 +81,48 @@ func TestWorkerAuth(ctx context.Context, t *testing.T, conn *db.DB, worker *Work
...
+func TestWorker(t *testing.T, conn *db.DB, wrapper wrapping.Wrapper, opt ...Option) *Worker {
 	t.Helper()
 	rw := db.New(conn)
 	kms := kms.TestKms(t, conn, wrapper)
 	serversRepo, err := NewRepository(rw, rw, kms)
 	require.NoError(t, err)
+	ctx := context.Background()
 
-	id, err := newWorkerId(context.Background())
+	namePart, err := newWorkerId(context.Background())
```
Comment (A-440Hz) [inline internal/servers/testing.go:95]: why is this context different from the one used for `UpdateWorker` at the end?

File: internal/daemon/controller/handlers/workers/worker_service_test.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+		},
+		{
+			name: "Filter to a single workers",
+			req:  &pbs.ListWorkersRequest{ScopeId: "global", Recursive: true, Filter: `"/item/name"=="worker2"`},
+			res: &pbs.ListWorkersResponse{
+				Items: wantWorkers[2:3],
+			},
+		},
+		{
+			name: "Filter to a 2 workers",
```
Comment (irenarindos) [inline internal/daemon/controller/handlers/workers/worker_service_test.go:176]:
```suggestion
			name: "Filter to 2 workers",
```

File: internal/proto/controller/api/resources/workers/v1/worker.proto (hunk 1 above)
Comment (irenarindos) [reply internal/proto/controller/api/resources/workers/v1/worker.proto:22]: Does the version field on the workerAggregate need to be removed as well?

**Review by irenarindos — APPROVED** [review]

**Pull request merged**